src/utils/formUtil.ts

```diff
@@ -330,22 +330,27 @@ export function eachComponent(
     }
     const info = componentInfo(component);
     let noRecurse = false;
-
+    Object.defineProperty(component, 'path', {
+      enumerable: false,
+      writable: true,
+      value: componentPath(component, path)
+    });
     // Keep track of parent references.
     if (parent) {
       // Ensure we don't create infinite JSON structures.
       Object.defineProperty(component, 'parent', {
         enumerable: false,
         writable: true,
-        value: { ...parent }
+        value: JSON.parse(JSON.stringify(parent))
```

Comment:
This is going to stringify the full Component instance.  I am not sure if this is what you are wanting...  We may consider just adding the parent component, but that would require investigation to figure out what is using this and then see if it needs any methods of the component instances.

Reply:
That is not a component instance, but just component JSON, and I  suppose there is nothing wrong with cloning the whole object

Reply:
That is not a component instance, but just component JSON, and I suppose there is nothing wrong with cloning the whole object